src/compiler/compile-props.js

```diff
@@ -120,6 +125,28 @@ module.exports = function compileProps (el, propOptions) {
 }
 
 /**
+ * Check existance of an attribute with binding syntax.
+ *
+ * @param {Element} el
+ * @return {String} attr
+ */
+
+function checkBindingAttr (el, attr) {
+  if (attr === 'class') {
+    return null
+  }
+
+  return (
+    el.getAttribute(':' + attr) ||
```

Comment:
Looks like we can just use `el.hasAttribute` here and the returned value will be a boolean.